dev/ppc64le/src/poly_tomont.S

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) The mlkem-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+#
+# Copyright 2025- IBM Corp.
+#
+#===================================================================================
+# Written by Danny Tsen <dtsen@us.ibm.com>
+#
+
+# Poly_tomont: Inplace conversion of all coefficients of a polynomial
+#              from normal domain to Montgomery domain
+#
+# Arguments:*r: pointer to input/output polynomial
+#
+
+#include "../../../common.h"
+
+#define V1353	0
+#define V_QINV  2
+#define V_NMKQ  5
+
+.machine "any"
+.text
+
+#
+# montgomery_reduce
+# t = a * QINV
+# t = (a - (int32_t)t*_MLKEM_Q) >> 16
+#
+#-----------------------------------
+# MREDUCE_4X(_v0, _v1, _v2, _v3)
+#
+.macro MREDUCE_4X _v0 _v1 _v2 _v3
+	lxvd2x	32+13, 0, 3
+	addi	3, 3, 16
+	lxvd2x	32+18, 0, 3
+	addi	3, 3, 16
+	lxvd2x	32+23, 0, 3
+	addi	3, 3, 16
+	lxvd2x	32+7, 0, 3
+	addi	3, 3, 16
+
+	vmladduhm 15, 13, V1353, 3
+	vmladduhm 20, 18, V1353, 3
+	vmladduhm 25, 23, V1353, 3
+	vmladduhm 9, 7, V1353, 3
+
+	vmhraddshs 14, 13, V1353, 3
+	vmhraddshs 19, 18, V1353, 3
+	vmhraddshs 24, 23, V1353, 3
+	vmhraddshs 8, 7, V1353, 3
+
+	vmladduhm 15, 15, V_QINV, 3
+	vmladduhm 20, 20, V_QINV, 3
+	vmladduhm 25, 25, V_QINV, 3
+	vmladduhm 9, 9, V_QINV, 3
+
+	vmhraddshs 15, 15, V_NMKQ, 14
+	vmhraddshs 20, 20, V_NMKQ, 19
+	vmhraddshs 25, 25, V_NMKQ, 24
+	vmhraddshs 9, 9, V_NMKQ, 8
+
+	vsrah \_v0, 15, 4		# >> 1
+	vsrah \_v1, 20, 4		# >> 1
+	vsrah \_v2, 25, 4		# >> 1
+	vsrah \_v3, 9, 4		# >> 1
+.endm
+
+.macro Write_8X
+	stxvd2x	32+27, 4, 3
+	stxvd2x	32+28, 5, 3
+	stxvd2x	32+29, 6, 3
+	stxvd2x	32+30, 7, 3
+	stxvd2x	32+13, 8, 3
+	stxvd2x	32+18, 9, 3
+	stxvd2x	32+23, 10, 3
+	stxvd2x	32+7, 11, 3
+.endm
+
+.align 4
```

Comment:
It seems that on PPC this means 16-byte alignment. Is this intentional, or would `.balign 4` -- 4 byte alignment -- do? The latter is what we use in all other assembly files and what `simpasm` would use by default.

Reply:
Sorted. We use 16 byte alignment on PPC.